### PR TITLE
feat: adding header handling to virtualized grids

### PR DESCRIPTION
**Pull request by JulienIzz**

### CODE REVIEW

File: packages/lib/src/spatial-navigation/components/virtualizedGrid/helpers/convertToGrid.ts [hunk 1]
```diff
@@ -6,10 +6,15 @@ import { NodeOrientation } from '../../../types/orientation';
...
   data: T[],
   numberOfColumns: number,
+  header?: JSX.Element,
 ): GridRowType<T>[] => {
   const rows: T[][] = chunk(data, numberOfColumns);
 
-  return rows.map((items, index) => ({ items, index }));
+  return rows.map((items, index) => {
+    //We do this to have index taking into account the header
+    if (header) index++;
```
Comment (pierpo) [inline packages/lib/src/spatial-navigation/components/virtualizedGrid/helpers/convertToGrid.ts:15]: this is mutating a parameter of the function, it's never a good idea 🙈 
I don't even understand why this works and has an effect on further iterations haha

File: packages/lib/src/spatial-navigation/components/virtualizedGrid/SpatialNavigationVirtualizedGrid.tsx [hunk 2]
```diff
@@ -199,17 +220,26 @@ export const SpatialNavigationVirtualizedGrid = typedMemo(
...
     );
+    const renderHeaderThenRows = useCallback(
+      ({ item }: { item: GridRowType<T> | JSX.Element }) => {
+        if (React.isValidElement(item)) {
+          return item;
+        }
+        return renderRow({ item: item as GridRowType<T> });
+      },
+      [renderRow],
+    );
```
Comment (JulienIzz) [inline packages/lib/src/spatial-navigation/components/virtualizedGrid/SpatialNavigationVirtualizedGrid.tsx:231]: @pierpo not really sure about this one in particular (and its little brother up there too)

File: packages/lib/src/spatial-navigation/components/virtualizedGrid/helpers/convertToGrid.ts (hunk 1 above)
Comment (JulienIzz) [reply packages/lib/src/spatial-navigation/components/virtualizedGrid/helpers/convertToGrid.ts]: Done :D

**Review by pierpo — APPROVED** [review]
🎉

**Pull request merged**